contracts/margin/external/BucketLender/BucketLender.sol

```diff
@@ -0,0 +1,1242 @@
+        );
+
+        MarginCommon.LoanOffering memory loanOffering = parseLoanOffering(
+            addresses,
+            values256,
+            values32,
+            signature
+        );
+
+        // CHECK ADDRESSES
+        assert(loanOffering.owedToken == OWED_TOKEN);
+        assert(loanOffering.heldToken == HELD_TOKEN);
+        assert(loanOffering.payer == address(this));
+        assert(loanOffering.owner == address(this));
+        require(
+            loanOffering.taker == address(0),
+            "BucketLender#verifyLoanOffering: loanOffering.taker is non-zero"
+        );
+        require(
+            loanOffering.feeRecipient == address(0),
+            "BucketLender#verifyLoanOffering: loanOffering.feeRecipient is non-zero"
+        );
+        require(
+            loanOffering.positionOwner == address(0),
+            "BucketLender#verifyLoanOffering: loanOffering.positionOwner is non-zero"
+        );
+        require(
+            loanOffering.lenderFeeToken == address(0),
+            "BucketLender#verifyLoanOffering: loanOffering.lenderFeeToken is non-zero"
+        );
+        require(
+            loanOffering.takerFeeToken == address(0),
+            "BucketLender#verifyLoanOffering: loanOffering.takerFeeToken is non-zero"
+        );
+
+        // CHECK VALUES256
+        require(
+            loanOffering.rates.maxAmount == MathHelpers.maxUint256(),
+            "BucketLender#verifyLoanOffering: loanOffering.maxAmount is incorrect"
+        );
+        require(
+            loanOffering.rates.minAmount == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.minAmount is non-zero"
+        );
+        require(
+            loanOffering.rates.minHeldToken == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.minHeldToken is non-zero"
+        );
+        require(
+            loanOffering.rates.lenderFee == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.lenderFee is non-zero"
+        );
+        require(
+            loanOffering.rates.takerFee == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.takerFee is non-zero"
+        );
+        require(
+            loanOffering.expirationTimestamp == MathHelpers.maxUint256(),
+            "BucketLender#verifyLoanOffering: expirationTimestamp is incorrect"
+        );
+        require(
+            loanOffering.salt == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.salt is non-zero"
+        );
+
+        // CHECK VALUES32
+        require(
+            loanOffering.callTimeLimit == CALL_TIMELIMIT,
+            "BucketLender#verifyLoanOffering: loanOffering.callTimelimit is incorrect"
+        );
+        require(
+            loanOffering.maxDuration == MAX_DURATION,
+            "BucketLender#verifyLoanOffering: loanOffering.maxDuration is incorrect"
+        );
+        assert(loanOffering.rates.interestRate == INTEREST_RATE);
```

Comment:
move this up top with the other asserts

Reply:
Also just want to confirm this is set by the base protocol on increase - I believe it is

Reply:
yes